x-pack/plugins/observability_solution/infra/common/http_api/custom_dashboards_api.ts

```diff
@@ -12,21 +12,22 @@ const AssetTypeRT = rt.type({
   assetType: ItemTypeRT,
 });
 
-const CustomDashboardRT = rt.intersection([
-  AssetTypeRT,
-  rt.type({
-    dashboardIdList: rt.array(rt.string),
-  }),
-  rt.partial({
-    kuery: rt.string,
-  }),
-]);
+const PayloadRT = rt.type({
+  dashboardSavedObjectId: rt.string,
+  dashboardFilterAssetIdEnabled: rt.boolean,
+});
+
+const InfraCustomDashboardRT = rt.intersection([AssetTypeRT, PayloadRT]);
+
+const dashboardSavedObjectIdRT = rt.type({
+  dashboardSavedObjectId: rt.string,
+});
 
 /**
  GET endpoint
 */
 export const InfraGetCustomDashboardsRequestParamsRT = AssetTypeRT;
-export const InfraGetCustomDashboardsResponseBodyRT = CustomDashboardRT;
+export const InfraGetCustomDashboardsResponseBodyRT = rt.array(InfraCustomDashboardRT);
```

Comment:
In order for the suggestions to work, we'd have to return the id of the custom dashboard saved object too.

Reply:
Added to the type `InfraCustomDashboardRT`